src/umi_tools/umi_tools_extract/config.vsh.yaml

```
@@ -0,0 +1,197 @@
+name: umi_tools_extract
+namespace: umi_tools
+description: |
+  Flexible removal of UMI sequences from fastq reads.
+  UMIs are removed and appended to the read name. Any other barcode, for example a library barcode,
+  is left on the read. Can also filter reads by quality or against a whitelist.
+keywords: [ extract, umi-tools, umi, fastq ]
+links:
+  homepage: https://umi-tools.readthedocs.io/en/latest/
+  documentation: https://umi-tools.readthedocs.io/en/latest/reference/extract.html
+  repository: https://github.com/CGATOxford/UMI-tools
+references: 
+  doi: 10.1101/gr.209601.116
+license: MIT
+
+argument_groups:
+
+  - name: Input
+    arguments: 
+    - name: --paired
+      type: boolean_true
+      description: Paired fastq files. If option is set, two input files are expected.
+    - name: --input
+      type: file
+      required: true
+      description: Input fastq files, either one or two (paired).
+      example: sample.fastq
+    - name: --read2_in
+      type: file
+      description: Filename for read pairs.
```

This combination of arguments and descriptions confuses me a bit.

It seems to me like the `--paired` argument is not an argument of `umi_tools extract`. If I'm not mistaken, you could determine whether we're doing a paired or an unpaired run based on whether `--read2_in` is defined or not.

However, based on the description of `--paired` and `--input`, it would appear as if `--read_in` should not exist and `--input` should be `multiple: true`.

If I look at the actual documentation in `help.txt`, I feel like this would make sense:

```suggestion
    - name: --input
      type: file
      required: true
      description: File containing the input data.
      example: sample.fastq
    - name: --read2_in
      type: file
      required: false
      description: File containing the input data for the R2 reads (if paired). If provided, a <list of other required arguments> need to be provided.
      example: sample_R2.fastq
```

↑ I adapted the description of `read2_in` for the sake of clarity.

---

Yes this seems good, I will adapt the script to fit this :)